.github/workflows/auditbeat-macos-unit-tests.yml

```diff
@@ -0,0 +1,55 @@
+name: auditbeat-macos-unit-tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+    paths:
+      - auditbeat/
+      # OSS
+      - go.mod
+      - pytest.ini
+      - dev-tools/
+      - libbeat/**
+      - testing/**
+      # Workflow
+      - .github/workflows/auditbeat-macos-unit-tests.yml
+      - .github/workflows/actions/unit-test-composite-action/action.yml
+
+permissions:
+  contents: read
+
+## Concurrency only allowed in the main branch.
+## So old builds running for old commits within the same Pull Request are cancelled
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  unit-tests-push:
+    name: Unit tests ${{ matrix.os }}
+    if: github.event_name == 'push'
+    strategy:
+      fail-fast: false
+      max-parallel: 10
+      matrix:
+        os: [macos-13, macos-14]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ./.github/workflows/actions/unit-test-composite-action
+        with:
+          beat: 'auditbeat'
+
+  unit-tests-pr:
+    name: Unit tests macos-15
+    if: github.event_name == 'pull_request'
+    runs-on: [ macos-15 ]
```

Comment:
Do you think that adding a [timeout](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes) could be beneficial here?